src/target/riscv/riscv.c

```diff
@@ -620,6 +622,15 @@ static int find_first_trigger_by_id(struct target *target, int unique_id)
 static int set_trigger(struct target *target, unsigned int idx, riscv_reg_t tdata1, riscv_reg_t tdata2,
 	riscv_reg_t tdata1_ignore_mask)
 {
+	RISCV_INFO(r);
+	assert(r->reserved_triggers);
```

Comment:
This assertion will become unnecessary if `reserved_triggers` are allocated statically.